images/archive/importer.go

```diff
@@ -381,3 +385,29 @@ func writeManifest(ctx context.Context, cs content.Ingester, manifest interface{
 
 	return desc, nil
 }
+
+func detectLayerMediaType(ctx context.Context, store content.Store, desc ocispec.Descriptor) (string, error) {
+	var mediaType string
+	// need to parse existing blob to use the proper media type
+	bytes := make([]byte, 10)
+	ra, err := store.ReaderAt(ctx, desc)
+	if err != nil {
+		return "", fmt.Errorf("failed to read content store to detect layer media type: %w", err)
+	}
+	defer ra.Close()
+	_, err = ra.ReadAt(bytes, 0)
+	if err != nil && err != io.EOF {
+		return "", fmt.Errorf("failed to read header bytes from layer to detect media type: %w", err)
+	}
+	if err == io.EOF {
+		// in the case of an empty layer then the media type should be uncompressed
+		return images.MediaTypeDockerSchema2Layer, nil
+	}
```

Comment:
nit: this is really subjective, but I'd avoid repeating `io.EOF`, e.g.

```suggestion
	if err != nil  {
		if err == io.EOF {
			// in the case of an empty layer then the media type should be uncompressed
			return images.MediaTypeDockerSchema2Layer, nil
		}
		return "", fmt.Errorf("failed to read header bytes from layer to detect media type: %w", err)
	}
```

But looks good, LGTM